libgadget/winds.c

```diff
@@ -396,17 +476,28 @@ sfr_wind_weight_ngbiter(TreeWalkQueryWind * I,
         double wk = 1.0;
         O->TotalWeight += wk * P[other].Mass;
     }
-
+    
+    int i;
     if(P[other].Type == 1) {
         const double atime = WIND_GET_PRIV(lv->tw)->Time;
-        if(r > I->DMRadius) return;
-        O->Ngb ++;
-        int d;
-        for(d = 0; d < 3; d ++) {
-            /* Add hubble flow; FIXME: this shall be a function, and the direction looks wrong too. */
-            double vel = P[other].Vel[d] + WIND_GET_PRIV(lv->tw)->hubble * atime * atime * dist[d];
-            O->V1sum[d] += vel;
-            O->V2sum += vel * vel;
+        for(i = 0; i < O->maxcmpte; i++){
+            if(r < I->DMRadius[i]){
+                O->Ngb[i] += 1;
+                int d;
+                for(d = 0; d < 3; d ++) {
+                    /* Add hubble flow; FIXME: this shall be a function, and the direction looks wrong too. */
+                    double vel = P[other].Vel[d] + WIND_GET_PRIV(lv->tw)->hubble * atime * atime * dist[d];
+                    O->V1sum[i][d] += vel;
+                    O->V2sum[i] += vel * vel;
+                }
+            }
+        }
+    }
+    
+    for(i = 0; i<NWINDHSML; i++){
+        if(O->Ngb[i] > NUMDMNGB){
+            O->maxcmpte = i+1;
```

Comment:
For this to be effective at reducing computation time, you would need to do
            iter->base.Hsml = I->Hsml[i];
and use 
tw->visit = treewalk_visit_nolist_ngbiter;
for the treewalk, as in metal_return.c
The reason is that the standard ngbiter function gathers all neighbours in the Ngblist before evaluating any of them, whereas the modified one evaluates them as it finds them.

Reply:
Thanks, I've fixed that part. The cpu time performance is similar as before. Maybe the new star is too little in my small test run to exhibit the effect.

Reply:
Sounds very likely. Thanks though!